lib/plugins/extension/helper/extension.php

```diff
@@ -638,7 +638,22 @@ public function installOrUpdate() {
      */
     public function uninstall() {
         $this->purgeCache();
-        return io_rmdir($this->getInstallDir(), true);
+
+        $data = array(
+            'extension' => $this,
+            'action' => 'uninstall'
+        );
+        $trigger = new Doku_Event('PLUGIN_EXTENSION_CHANGE', $data);
+
+        if($trigger->advise_before()) {
+            if(io_rmdir($this->getInstallDir(), true)) {
+                $trigger->advise_after();
+                return true;
+            } else {
+                return false;
+            }
+        }
+        return (bool) $trigger->result;
```

Comment:
This cast to `bool` is very different from the semantics of the `enable` and `disable` events where a string is an error message but here a non-empty string means success. It should be easy to make this more consistent also in the extension manager itself.